src/core/components/ChatHistory.tsx

```diff
@@ -154,12 +149,17 @@ const HistoryItem = memo(
       [id],
     );
 
+    const [didEditTitle, setDidEditTitle] = useState(false);
+
```

Comment:
I'm testing this locally on main and I can't quite reproduce not getting the edit to save. 

https://github.com/user-attachments/assets/c842b5d8-e1fe-4343-8a94-95f210755ed2

Reply:
Sounds good - this ^ was on main.

Reply:
Repro'd, tested, and fixed on #378